fix: cli encoding issue

oliverqx opened the pull request:
## description
pr to fix cli encoding issue

related issue: #1247

oliverqx commented:
@louis030195 so it took me some time to replicate consistently. I'm not sure whats causing the issue. The behaviour is wild, so its very hard to understand. 

1.  Encoding issue does not always occur but once it starts happening it doesn't go away.
2.  Usually starts happening after running @screenpipe/dev create and ctrl+c in the middle of it ?
3.  Encoding breaks ONLY for the components add command, which I also do not understand..
4. I thought it was related to prompts, the package, but even by using inquirer it does not encode well. 

Only way I've been able to get it working, is by switching from brocli to Commander, so I wanted to know if you'd be okay with me doing that. 

Apart from fixing the issue, commander has a wider community support. Their last release was only 2 weeks ago. It has 110M weekly downloads (brocli less than 100K). I've also been using Commander for years now and it has never presented weird encoding issues like this one

louis030195 commented:
> @louis030195 so it took me some time to replicate consistently. I'm not sure whats causing the issue. The behaviour is wild, so its very hard to understand.
> 
> 1. Encoding issue does not always occur but once it starts happening it doesn't go away.
> 2. Usually starts happening after running @screenpipe/dev create and ctrl+c in the middle of it ?
> 3. Encoding breaks ONLY for the components add command, which I also do not understand..
> 4. I thought it was related to prompts, the package, but even by using inquirer it does not encode well.
> 
> Only way I've been able to get it working, is by switching from brocli to Commander, so I wanted to know if you'd be okay with me doing that.
> 
> Apart from fixing the issue, commander has a wider community support. Their last release was only 2 weeks ago. It has 110M weekly downloads (brocli less than 100K). I've also been using Commander for years now and it has never presented weird encoding issues like this one

would be great to switch to commander

oliverqx commented:
@louis030195 wdyt about adding some tests? i see theres a folder already but there are no tests yet

louis030195 commented:
is it mergeable?

oliverqx commented:
@louis030195 it is yes. changes:

1. commander migration
2. implemented logger
3. refactored create command
4. create pipe command
5. create app command
6. using simpleGit now for better nodejs\process support

oliverqx commented:
as always, i've tested it thoroughly. Will keep an eye out for edge cases/bugs but I'm fairly sure its alright

oliverqx commented:
was thinking we could refactor DX into

1. create pipe [init,register,publish]
2. create app [init, x] 

x could be some app template specific command to run

louis030195 commented:
<img width="763" alt="image" src="https://github.com/user-attachments/assets/c310bca2-4f1b-45d2-bf92-80a47d0c895a" />

louis030195 commented:
@screenpipe/dev app create [name]


make sense

oliverqx commented:
@louis030195 i rolledback create command refactor. 

<img width="704" alt="Screenshot 2025-02-05 at 16 26 29" src="https://github.com/user-attachments/assets/de3e2825-84a4-487b-ba9f-f6f89ce07ba0" />

louis030195 commented:
nice

what about 

@screenpipe/dev pipe [create,register,publish]

oliverqx commented:
@louis030195 just refactored!

<img width="535" alt="Screenshot 2025-02-05 at 21 29 44" src="https://github.com/user-attachments/assets/11f7d277-1191-453b-b8f6-b7dfe62e9091" />

louis030195 commented:
nice!

louis030195 commented:
/tip $150 @oliverqx

louis030195 commented:
https://github.com/user-attachments/assets/98b4b794-7180-4340-b773-d24218729ff6




@oliverqx for some reason i cannot use the published CLI, works in dev mode tho

oliverqx commented:
@louis030195 looks like i missed the shebang, usually run it as node ./dist/index.js. just opened a pr to fix that!
